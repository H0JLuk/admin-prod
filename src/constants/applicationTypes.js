const applicationTypes = [
    {type: 'IOS'},
    {type: 'ANDROID'},
    {type: 'OTHER'}
];

export default applicationTypes;