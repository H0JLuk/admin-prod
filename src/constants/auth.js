export const LOGIN_TYPE = {
    PASSWORD: 'PASSWORD',
    // SBOL_PRO: 'SBOL_PRO',
    SBER_USER_ID: 'SBER_USER_ID',
};