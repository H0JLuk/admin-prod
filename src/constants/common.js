export const showCount = { formatter: ({ count, maxLength }) => `Осталось символов ${maxLength - count}` };

/* '^(https?:\\/\\/)?'+ // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
    '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
    '(\\:\\d+)?(\\/[-a-z\\d%_.~+{}]*)*'+ // port and path
    '(\\?[;&a-z\\d%_.~+=-{}]*)?'+ // query string
    '(\\#[-a-z\\d_]*)?$','i'); // fragment locator */
export const URL_REGEXP = /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(:\d+)?(\/[-A-Za-z\d%_.~+#]*)*(\?[;&a-z\d%_.~+=-{}]*)?(#[-a-z\d_]*)?$/;
