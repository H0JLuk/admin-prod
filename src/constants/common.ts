import { TextAreaProps } from 'antd/lib/input';

export const showCount: TextAreaProps['showCount'] = {
    formatter: ({ count, maxLength }) => `Осталось символов ${maxLength! - count}`,
};

export const DEFAULT_SLEEP_TIME = 500; // in milliseconds

export enum DIRECTION {
    ASC = 'ASC',
    DESC = 'DESC',
}
