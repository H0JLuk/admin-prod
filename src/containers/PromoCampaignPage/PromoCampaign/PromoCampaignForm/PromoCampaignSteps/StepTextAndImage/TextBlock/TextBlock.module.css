.textArea {
    margin-bottom: 10px;
    max-height: 80px;
}

.detail {
    color:#b5b5b5;
}

.textBlock :global(.ant-form-item) {
    margin-bottom: 0;
}