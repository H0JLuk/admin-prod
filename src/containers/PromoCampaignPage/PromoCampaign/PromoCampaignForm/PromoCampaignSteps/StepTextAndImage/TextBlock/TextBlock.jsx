import React from 'react';
import { Form, Input } from 'antd';
import { showCount } from '../../../../../../../constants/common';

import styles from './TextBlock.module.css';

const TextBlock = ({
    title,
    placeholder,
    rows = 3,
    maxLength = null,
    rules,
    name,
    initialValue = '',
}) => {

    return (
        <div className={ styles.textBlock }>
            <Form.Item
                label={ title }
                name={ name }
                initialValue={ initialValue }
                rules={ rules }
            >
                <Input.TextArea
                    maxLength={ maxLength || null }
                    placeholder={ placeholder }
                    rows={ rows }
                    className={ styles.textArea }
                    showCount={ maxLength && showCount }
                />
            </Form.Item>

        </div>
    );
};

export default TextBlock;
