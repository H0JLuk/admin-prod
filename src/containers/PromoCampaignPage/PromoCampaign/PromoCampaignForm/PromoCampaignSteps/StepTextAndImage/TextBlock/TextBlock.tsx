import React from 'react';
import { Form, FormItemProps, Input } from 'antd';
import { showCount } from '@constants/common';

import styles from './TextBlock.module.css';

export type TextBlockProps = {
    title: string;
    placeholder?: string;
    rows?: number;
    maxLength?: number;
    rules: FormItemProps['rules'];
    name: FormItemProps['name'];
    initialValue: string;
};

const TextBlock: React.FC<TextBlockProps> = ({
    title,
    placeholder,
    rows = 3,
    maxLength = null,
    rules,
    name,
    initialValue = '',
}) => (
    <div className={styles.textBlock}>
        <Form.Item
            label={title}
            name={name}
            initialValue={initialValue}
            rules={rules}
            validateFirst
        >
            <Input.TextArea
                maxLength={(maxLength || null) as number}
                placeholder={placeholder}
                rows={rows}
                className={styles.textArea}
                showCount={maxLength ? showCount : false}
            />
        </Form.Item>

    </div>
);

export default TextBlock;
