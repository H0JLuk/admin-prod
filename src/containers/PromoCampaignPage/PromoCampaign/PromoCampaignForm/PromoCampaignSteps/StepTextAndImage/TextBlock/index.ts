import TextBlock from './TextBlock';

export type { TextBlockProps } from './TextBlock';

export default TextBlock;
