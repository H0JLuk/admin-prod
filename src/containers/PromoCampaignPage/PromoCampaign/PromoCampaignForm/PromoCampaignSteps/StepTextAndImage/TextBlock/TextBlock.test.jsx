import { shallow } from 'enzyme';
import React from 'react';
import { showCount } from '../../../../../../../constants/common';
import TextBlock from './TextBlock';

describe('<TextBlock /> tests', () => {
    const props = {
        title: 'Условия',
        placeholder: 'Текст условий',
        rows: 3,
        maxLength: 60,
        rules: [
            {
                message: 'Заполните поле',
                required: true
            },
        ],
        name: ['texts', 'RULES'],
        initialValue: '',
    };
    const TextBlockItem = shallow(<TextBlock { ...props } />);

    it('TextBlock snapshot', () => {
        expect(TextBlockItem.debug()).toMatchSnapshot();
    });

    it('should show count', () => {
        expect(TextBlockItem.find('ForwardRef').prop('showCount')).toEqual(showCount);
    });
});
