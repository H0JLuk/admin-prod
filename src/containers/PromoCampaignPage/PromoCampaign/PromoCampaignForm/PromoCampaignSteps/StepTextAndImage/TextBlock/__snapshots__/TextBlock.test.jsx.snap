// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBlock /> tests TextBlock snapshot 1`] = `
"<div className=\\"textBlock\\">
  <FormItem label=\\"Условия\\" name={{...}} initialValue=\\"\\" rules={{...}} validateTrigger=\\"onSubmit\\" validateFirst={true}>
    <ForwardRef maxLength={60} placeholder=\\"Текст условий\\" rows={3} className=\\"textArea\\" showCount={{...}} />
  </FormItem>
</div>"
`;
